src/scrypt.rs

```diff
@@ -32,8 +32,8 @@ pub fn derive_key(pass: &Password, salt: &[u8; 32], n: u32, p: u32, r: u32) -> R
     }
 
     let mut derived_key = vec![0u8; KEY_LENGTH];
-    let scrypt_params = ScryptParams::new(log_n, r, p);
-    scrypt(pass.as_bytes(), salt, &scrypt_params, &mut derived_key);
+    let scrypt_params = ScryptParams::new(log_n, r, p).unwrap();
```

Comment:
You should also handle the error related to parameters

Reply:
I modified it. Thank you : )